typescript/src/util.ts

```diff
@@ -159,12 +159,15 @@ export function getResourceNameByPattern(pattern: string): string {
       const nextEle = elements[0];
       if (nextEle.match(/{[a-zA-Z_]+(?:=.*?)?}/g)) {
         elements.shift();
-        name.push(
-          nextEle.substring(
-            1,
-            nextEle.includes('=') ? nextEle.indexOf('=') : nextEle.length - 1
-          )
-        );
+        const params = nextEle.match(/{[a-zA-Z_]+(?:=.*?)?}/g);
+        if (params!.length === 1) {
+          name.push(getChuckName(nextEle));
+        } else {
+          // For non-slash resource 'organization/{organization}/tasks/{task_id}{task_name}/result'
+          // Take parameters that match pattern [{task_id}, {task_name}] and combine them as part of the name.
+          const params = nextEle.match(/{[a-zA-Z_]+(?:=.*?)?}/g);
+          name.push(params?.map(p => getChuckName(p))?.join('_'));
```

Comment:
We don't really need a separate function since you already have a regex here. Let's just improve your regex by putting everything _other_ than what we need into positive look-behind and look-ahead:

```
> s = "/foo/{ab}~{cd=zzz}123{efg}"
'/foo/{ab}~{cd=zzz}123{efg}'
> s.match(/(?<=\{).*?(?=(?:=.*?)?\})/g)
[ 'ab', 'cd', 'efg' ]
```

Looking closer at the regex, `/(?<=\{).*?(?=(?:=.*?)?\})`, the first opening bracket is in the positive look-behind `(?<=    \{      )`, then the part we care about follows  `.*?`  (can be changed to `[a-zA-Z_]+` if you prefer), and then a positive look-ahead eats optional trailing `=...` and the closing bracket: `(?=      (?:=.*?)?      \}    )`.

Reply:
Thank you! learned more about regex matching (I was not skilled at using ? mark)